internal/util/K8sUtil.go

```diff
@@ -165,6 +179,82 @@ func (impl K8sUtil) GetConfigMap(namespace string, name string, client *v12.Core
 	}
 }
 
+const watcherRestart = "restart event watcher"
+
+func (impl K8sUtil) WatchConfigMap(namespace string, dir string, client kubernetes.Interface) (string, string, []byte, error) {
+	api := client.CoreV1().ConfigMaps(namespace)
+	configMaps, err := api.List(metav1.ListOptions{})
+	if err != nil {
+		return "", "", nil, err
+	}
+
+	resourceVersion := configMaps.ListMeta.ResourceVersion
+	timeout := int64(1800)
+	watcher, err := api.Watch(metav1.ListOptions{ResourceVersion: resourceVersion, TimeoutSeconds: &timeout})
+	if err != nil {
+		return "", "", nil, err
+	}
+
+	ch := watcher.ResultChan()
+
+	for {
+		select {
+		case event, ok := <-ch:
+			if !ok {
+				// the channel got closed, so we need to restart
+				return "", "", nil, error2.New(watcherRestart)
+			}
+			configMaps, ok := event.Object.(*coreV1.ConfigMap)
+			if !ok {
+				return "", "", nil, nil
+			}
+			annotations, ok := configMaps.Annotations["charts.devtron.ai/data"]
```

Comment:
decouple it, dont do extraction here, take a callback function and do extraction over there